File: backends/java/zappy-adapters/src/main/java/com/zappymart/adapters/mail/ConsoleMailer.java
package com.zappymart.adapters.mail;

import com.zappymart.application.ports.Mailer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public final class ConsoleMailer implements Mailer {

    private static final Logger LOGGER = LoggerFactory.getLogger(ConsoleMailer.class);

    @Override
    public void send(MailMessage message) {
        LOGGER.info("Mail to {} with subject {}{}{}", message.recipient(), message.subject(),
                System.lineSeparator(), message.body());
    }
}

File: backends/kotlin/zappy-adapters/src/main/kotlin/nl/zappymart/adapters/mail/ConsoleMailer.kt
package nl.zappymart.adapters.mail

import nl.zappymart.application.ports.Mailer
import nl.zappymart.application.ports.OrderConfirmation
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class ConsoleMailer : Mailer {

    private val log = LoggerFactory.getLogger(ConsoleMailer::class.java)

    override fun send(confirmation: OrderConfirmation) {
        log.info(
            "Order confirmation for order {} to {}, total {} {}",
            confirmation.orderNumber,
            confirmation.recipient.value,
            confirmation.total.amount,
            confirmation.total.currency,
        )
    }
}
